src/Generator.h

```diff
@@ -552,6 +554,56 @@ class GeneratorParam_MachineParams : public GeneratorParamImpl<T> {
     }
 };
 
+class GeneratorParam_LoopLevel : public GeneratorParamImpl<LoopLevel> {
+public:
+    GeneratorParam_LoopLevel(const std::string &name, const LoopLevel &value) : GeneratorParamImpl<LoopLevel>(name, value) {}
+
+    void set_from_string(const std::string &new_value_string) override {
+        if (new_value_string == "root") {
+            this->set(LoopLevel::root());
+        } else if (new_value_string == "inlined") {
+            this->set(LoopLevel::inlined());
+        } else {
+            user_error << "Unable to parse " << this->name << ": " << new_value_string;
+        }
+    }
+
+    std::string get_default_value() const override {
+        // This is dodgy but safe in this case: we want to
+        // see what the value of our LoopLevel is *right now*,
+        // so we make a copy and lock the copy so we can inspect it.
+        // (Note that ordinarily this is a bad idea, since LoopLevels
+        // can be mutated later on; however, this method is only
+        // called by the Generator infrastructure, on LoopLevels that
+        // will never be mutated, so this is really just an elaborate way
+        // to avoid runtime assertions.)
+        LoopLevel copy;
+        copy.set(this->value());
+        copy.lock();
+        if (copy.is_inlined()) {
+            return "LoopLevel::inlined()";
```

Comment:
This doesn't need to match the strings in set_from_string? (and below)

Reply:
Nope: this is used to generate C++ code in the stub, to fill in the C++ default values in the GeneratorParams struct.